atmo/jobs/models.py

```diff
@@ -406,17 +407,54 @@ def sync(self, info=None):
             ('ready_datetime', 'ready_at'),
             ('end_datetime', 'finished_at'),
         )
+        save_needed = False
+        date_fields_updated = False
+
         # set the various model fields to the value the API returned
         for api_field, model_field in model_field_map:
             field_value = info.get(api_field)
-            if field_value is None:
+            if field_value is None or field_value == getattr(self, model_field):
                 continue
             setattr(self, model_field, field_value)
+            save_needed = True
+
+            if model_field in ('started_at', 'ready_at', 'finished_at'):
+                date_fields_updated = True
+
+        if save_needed:
+            # If the job cluster terminated with error raise the alarm.
+            if self.status == Cluster.STATUS_TERMINATED_WITH_ERRORS:
+                transaction.on_commit(lambda: self.alert(info))
+            self.save()
+
+        if date_fields_updated:
+            if self.finished_at:
+                # When job is finished, record normalized instance hours.
+                hours = math.ceil(
+                    (self.finished_at - self.started_at).seconds / 60 / 60
+                )
```

Comment:
This looks like a copy/paste of the one in clusters app, can we refactor this into one method please?

Reply:
Yeah, true. I actually typed it by using the cluster one as reference. I should have noticed myself how exact they were.